internal/backend/db/dbgorm/secrets.go

```diff
@@ -0,0 +1,71 @@
+package dbgorm
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+)
+
+func (db *gormdb) SetSecret(ctx context.Context, name string, data map[string]string) error {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	s := scheme.Secret{Name: name, Data: jsonData}
+	result := db.db.WithContext(ctx).Create(&s)
+	if result.Error != nil {
+		return translateError(result.Error)
+	}
+	return nil
+}
+
+func (db *gormdb) GetSecret(ctx context.Context, name string) (map[string]string, error) {
+	// Why Take() and not First()? We don't care about order because at
+	// most a single record exists (we retrieve based on the primary key).
+	// Also, if the primary key isn't found, Take() returns an error
+	s := scheme.Secret{}
+	result := db.db.WithContext(ctx).Take(&s, "name = ?", name)
+	if result.Error != nil {
+		return nil, translateError(result.Error)
+	}
+
+	data := make(map[string]string)
+	if err := json.Unmarshal(s.Data, &data); err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func (db *gormdb) AppendSecret(ctx context.Context, name, token string) error {
+	// No need to combine both queries into a single transaction,
+	// because only the last one affects the data.
+	data, err := db.GetSecret(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	data[token] = time.Now().UTC().Format(time.RFC3339)
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	result := db.db.WithContext(ctx).Model(&scheme.Secret{}).Where("name = ?", name).Update("data", jsonData)
```

Comment:
You can just try to update and if rows affected is zero it means secret does not exists 

You can save a call to db

Reply:
Yes, but unfortunately I need `Get()`. This function adds the `token` input into the existing data (see line 51), it doesn't overwrite existing data.

That said, this reminds me that I need to make sure if this function is expected to succeed in the data is not found, i.e. create a new row is it doesn't exist. I'll update the unit tests either way.

As for using this table - I'll reference this PR in the next PR which will do that. Not merging it yet, because it's stacked over #69.

Reply:
So if this is a json data type and you want to append stuff you can still do it with one sql command in Postgres
No need to fetch the data before (same for delete if it's by key)